### ISSUE TITLE
WIzard of Wikipedia does not train (finetune)

### ISSUE THREAD

**Issue by shashankMadan-designEsthetics** [facebookresearch/ParlAI#2869]
**Bug description**
I have created a custom task `bankQA` to train on Wow. The training does not happen it gives `time_left:0s` and goes on for validation and testing.
```
this is my training code

from parlai.scripts.train_model import TrainModel

TrainModel.main(
    # similar to before
    task='bankQA',
    model='projects.wizard_of_wikipedia.generator.agents:EndToEndAgent',
    model_file='/tmp/end2end_generator/model',

    # initialize with a pretrained model
    init_model='zoo:wizard_of_wikipedia/end2end_generator/model',

    # arguments we get from the pretrained model.
    # Unfortunately, these must be looked up separately for each model.
    # eps
    num_epochs=2,
    dict_lower=True,
    dict_tokenizer='bpe',
    n_layers=5,
    n_heads=2,
    dropout=0.20,
    ffn_size=512,
    embedding_size=256,
    log_every_n_secs=10,
    validation_patience=12,
    validation_metric='ppl',
    validation_metric_mode='min',
    validation_every_n_epochs=0.5,
    n_positions=128,
    truncate=128,
    max_knowledge=32,
    knowledge_alpha=0.95,
    knowledge_truncate=32,
    learningrate=5e-4,
    warmup_updates=5000,
    clip=0.1,
    lr_scheduler='invsqrt',
    embedding_type='fasttext',
    beam_size=1,
    skip_generation=False,
    batchsize=64,
    # fp16=True
)

```
This is the training stats

Total parameters: 15,585,024 (15,519,488 trainable)
[ Loading existing model params from /tmp/end2end_generator/model ]
/Volumes/Data/ParlAI/parlai/nn/lr_scheduler.py:398: UserWarning: --lr-scheduler invsqrt requires a value for --invsqrt-lr-decay-gamma. Defaulting to set gamma to --warmup-updates value for backwards compatibility.
  '--lr-scheduler invsqrt requires a value for '
[creating task(s): bankQA]
 ~~ Loading from train.json ~~ 
[ training... ]
[ time:263.0s total_exs:2880 epochs:2880.0 time_left:0s ]
    clip  exs  gnorm  know_acc  know_chance  know_loss  loss       lr   ppl  token_acc  total_train_updates  tpb  updates
       1   64  .2135         1            1          0 10.42 4.55e-06 33492          0                   45 2304        1

[ num_epochs completed:2.0 time elapsed:263.72646021842957s ]

I have kept `num_epoch=2` and 10, 100 but it just seems to move on to the validation and testing phase.
Please share any solution if possible. THanks

**Comment by stephenroller** [comment]
It looks like what's happening is you're not deleting the file between reruns. As such, the `model_file` is RELOADED every time, in an attempt to _resume_ training. This is done because our own internal systems can be preempted, and so we need to be able to resume from training at every point.

To be clear: you should delete the model files between each run, or you should use a new filename every time.

**Comment by shashankMadan-designEsthetics** [comment]
So i changed the `model_file` to a new folder `model_file='/from_pretrained_wiki/model',` and it gives me this error

`Could not load the model due to a size mismatch in the embeddings. A common reason for this is trying to load a model trained with fp16 but loaded without fp16. Try adding --fp16 true or --force-fp16-tokens true.`

I did fp16 true but still the same

**Comment by stephenroller** [comment]
Try adding `dict_file="'zoo:wizard_of_wikipedia/end2end_generator/model.dict",`

**Comment by shashankMadan-designEsthetics** [comment]
> Try adding `dict_file='zoo:wizard_of_wikipedia/end2end_generator/model.dict',`

Gives the exact same error...

**Comment by stephenroller** [comment]
It works well on a clean install and empty model folder: [demo here](https://colab.research.google.com/drive/1Hw8cO97jMK7SrXbltCxX1u6jh3SMPACt?usp=sharing).

**Comment by shashankMadan-designEsthetics** [comment]
> It works well on a clean install and empty model folder: [demo here](https://colab.research.google.com/drive/1Hw8cO97jMK7SrXbltCxX1u6jh3SMPACt?usp=sharing).

Yep found out the problem its actually because I have a new dataset with only 56 words it was building new `dict` and hence the `embedding_size` was `56` but after `dict_file='zoo:wizard_of_wikipedia/end2end_generator/model.dict'` of pretrained model works fine shows time left in seconds but train's just the last epoch

**Comment by stephenroller** [comment]
Cool, seems like we can close this?

**Comment by shashankMadan-designEsthetics** [comment]
Yep!
